tools/codegen/cmd/featuregate-test-analyzer.go

```diff
@@ -456,8 +476,32 @@ func listTestResultFor(featureGate string, clusterProfiles sets.Set[string]) (ma
 	return results, nil
 }
 
+func filterVariants(featureGate string, variants []JobVariant, optional bool) []JobVariant {
```

Comment:
if we looked at everything by default, then I think this doesn't need the "optional" flag.